astropy/stats/circstats.py

```diff
@@ -0,0 +1,481 @@
+    >>> data = np.deg2rad(np.array([51, 67, 40, 109, 31, 358]))
+    >>> circmoment(data, p=2) # doctest: +FLOAT_CMP
+    (1.5881210029361645, 0.48004283892950717)
+
+    References
+    ----------
+    .. [1] S. R. Jammalamadaka, A. SenGupta. "Topics in Circular Statistics".
+       Series on Multivariate Analysis, Vol. 5, 2001.
+    .. [2] C. Agostinelli, U. Lund. "Circular Statistics from 'Topics in
+       Circular Statistics (2001)'". 2015.
+       <https://cran.r-project.org/web/packages/CircStats/CircStats.pdf>
+    """
+
+    phi = 0.0
+    if centered:
+        phi = circmean(data,w,axis)
+
+    return _angle(data,w,p,phi,axis), _length(data,w,p,phi,axis)
+
+def circcorrcoef(alpha, beta, w_alpha=None, w_beta=None, ax_alpha=None,
+                 ax_beta=None):
+    """ Computes the circular correlation coefficient between two array of
+    circular data.
+
+    Parameters
+    ----------
+    alpha : numpy.ndarray
+        Array of circular (directional) data measured in radians.
+    beta : numpy.ndarray
+        Array of circular (directional) data measured in radians.
+    w_alpha : numpy.ndarray, optional
+        In case of grouped data, the i-th element of ``w_alpha`` represents a
+        weighting factor for each group such that sum(w,axis) equals the number
+        of observations. See [1], remark 1.4, page 22, for detailed
+        explanation.
+    w_beta : numpy.ndarray, optional
+        See description of ``w_alpha``.
+    ax_alpha : int, optional
+        Axis along which circular correlation coefficients are computed.
+        The default is the compute the circular correlation coefficient of the
+        flattened array.
+    ax_ beta : int, optional
+        See description of ``ax_alpha``
+
+    Returns
+    -------
+    rho : float
+        Circular correlation coefficient
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from astropy.stats.circstats import circcorrcoef
+    >>> alpha = np.deg2rad(np.array([356, 97, 211, 232, 343, 292, 157, 302,
+    ...                              335, 302, 324, 85, 324, 340, 157, 238,
+    ...                              254, 146, 232, 122, 329]))
+    >>> beta = np.deg2rad(np.array([119, 162, 221, 259, 270, 29, 97, 292, 40,
+    ...                             313, 94, 45, 47, 108, 221, 270, 119, 248,
+    ...                             270, 45, 23]))
+    >>> circcorrcoef(alpha, beta) # doctest: +FLOAT_CMP
+    0.27046488267488311
+
+    References
+    ----------
+    .. [1] S. R. Jammalamadaka, A. SenGupta. "Topics in Circular Statistics".
+       Series on Multivariate Analysis, Vol. 5, 2001.
+    .. [2] C. Agostinelli, U. Lund. "Circular Statistics from 'Topics in
+       Circular Statistics (2001)'". 2015.
+       <https://cran.r-project.org/web/packages/CircStats/CircStats.pdf>
+    """
+
+    alpha = np.asarray(alpha)
+    beta = np.asarray(beta)
+
+    assert np.size(alpha, axis=ax_alpha) == np.size(beta, axis=ax_beta)
```

Comment:
I think raising `ValueError` with a more descriptive error message will be more useful to the users.